capsules/src/ble_advertising_driver.rs

```diff
@@ -964,41 +641,33 @@ where
         allow_num: usize,
         slice: Option<kernel::AppSlice<kernel::Shared, u8>>,
     ) -> ReturnCode {
-        match AllowType::from_usize(allow_num) {
-            Some(AllowType::BLEGap(gap_type)) => self.app
+        match allow_num {
+            // Advertisement buffer
+            0 => self.app
                 .enter(appid, |app, _| {
-                    if app.process_status != Some(BLEState::NotInitialized) {
-                        app.app_write = slice;
-                        app.set_gap_data(gap_type)
+                    app.adv_data = slice;
+                    if let ReturnCode::SUCCESS = app.generate_random_address(appid) {
```

Comment:
Is there any way for an app to determine what its BLE address is?

This isn't something that has to be addressed in this PR, but I think it makes sense to allow an app to know which address it has been assigned. If I was writing an app, I would have it print out what its address is so that a human can find data sent by it.

Reply:
If an app knows its `AppID` it could be determined since the address is deduced by:

```0xf0 | AppId | 0xf0```    (AppId is 4 bytes)

But, we should add functionality for an app to fetch its advertising address!

Reply:
Agreed, we should have a system call interface for that. We should also consider allowing the app to _set_ an advertising address. I think both should be part of a discussion of the _next_ iteration of the BLE interface (i.e. after this PR). Is that acceptable?